rancher2/resource_rancher2_cluster.go

```diff
@@ -755,6 +719,73 @@ func getClusterKubeconfig(c *Config, id, origconfig string) (*managementClient.G
 	}
 }
 
+func updateClusterMonitoring(d *schema.ResourceData, meta interface{}, newCluster Cluster) error {
+	client, err := meta.(*Config).ManagementClient()
```

Comment:
The use of `meta.(*Config).ManagementClient()`  here creates an entirely new client which results in a few HTTP requests being made against the Rancher API. I think it would make more sense to reuse the client defined in the calling function, `resourceRancher2ClusterUpdate` on line 280. let me know what you think, or if I'm mistaken. I'm sure the additional http requests don't have too much impact, but every little bit helps